src/src/SessionStore.cpp

```diff
@@ -0,0 +1,206 @@
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownAuthorizationKeyException("Unknown authorization key");
+    }
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::registerClient(asio::ip::udp::endpoint endpoint)
+{
+    try {
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        return _udpSessions.at(endpoint);
+    } catch (std::out_of_range &e) {
+        auto sId = _findAvailableUdpSessionId();
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        _udpSessions.emplace(endpoint, sId);
+        return sId;
+    }
+}
+
+void polymorph::network::SessionStore::registerAuthoredClient(asio::ip::tcp::endpoint endpoint,
+                                                              polymorph::network::SessionId sessionId, polymorph::network::AuthorizationKey key)
+{
+    try {
+        std::unique_lock<std::mutex> lock(_tcpSessionsAuthorizationKeysMutex);
+        auto verifyKey = _tcpSessionsAuthorizationKeys.at(sessionId);
+        lock.unlock();
+        if (authorizationKey::areSame(verifyKey, key)) {
+            _tcpSessions[endpoint] = sessionId;
+        } else {
+            throw exceptions::UnauthorizedException("InvalidAuthorization key");
+        }
+    } catch (std::out_of_range &e) {
+        throw exceptions::UnknownAuthorizationKeyException("Unknown authorization key");
+    }
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::registerClient(asio::ip::tcp::endpoint endpoint)
+{
+    auto sId = _findAvailableTcpSessionId();
+    std::lock_guard<std::mutex> lock(_tcpSessionsMutex);
+
+    try {
+        _tcpSessions.at(endpoint);
+        throw exceptions::AlreadyRegisteredException("Client already connected");
+    } catch (std::out_of_range &e) {
+    }
+
+    _tcpSessions.emplace(std::move(endpoint), sId);
+    return sId;
+}
+
+void polymorph::network::SessionStore::removeClient(asio::ip::udp::endpoint endpoint)
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+
+    _udpSessions.erase(endpoint);
+}
+
+void polymorph::network::SessionStore::removeClient(asio::ip::tcp::endpoint endpoint)
+{
+    std::lock_guard<std::mutex> lock(_tcpSessionsMutex);
+
+    _tcpSessions.erase(endpoint);
+}
+
+bool polymorph::network::SessionStore::removeClient(polymorph::network::SessionId sessionid)
+{
+    std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+    std::lock_guard<std::mutex> lock2(_tcpSessionsMutex);
+
+    auto udpNb = std::erase_if(_udpSessions, [&sessionid](auto &pair) {
+        return pair.second == sessionid;
+    }) != 0;
+    auto tcpNb = std::erase_if(_tcpSessions, [&sessionid](auto &pair) {
+        return pair.second == sessionid;
+    }) != 0;
+    return udpNb != 0 || tcpNb != 0;
+}
+
+polymorph::network::SessionId polymorph::network::SessionStore::sessionOf(asio::ip::udp::endpoint endpoint)
+{
+    try {
+        std::lock_guard<std::mutex> lock(_udpSessionsMutex);
+        return _udpSessions.at(endpoint);
```

Comment:
Using contains or any conditional presence will be better than an entire try catch block